api/src/main/java/org/openmrs/api/impl/ObsServiceImpl.java

```diff
@@ -219,7 +219,7 @@ private void saveObsGroup(Obs obs, String changeMessage){
 
 	private void handleExistingObsWithComplexConcept(Obs obs) {
 		if (null != obs.getConcept() && obs.getConcept().isComplex()
-		        && null != obs.getComplexData().getData()) {
```

Comment:
Should we also leave the check for getData() not being null? Of course in addition to what you have just added. :)